### PR TITLE
`struct Rav1dColorPrimaries`: make a newtype wrapper

### PR DESCRIPTION
**Pull request by folkertdev**
I believe this is the final enum/struct in `headers.rs`

### CODE REVIEW

**Review by kkysen — COMMENTED** [review]
nit: I think your commit message for 6a97d59 has extra `` ` ``s.  And does this also get random/unknown values like the others?

**Comment by folkertdev** [comment]
fixed that typo, and yes it also does have the (seemingly) random values that are undefined by the spec.

File: include/dav1d/headers.rs [hunk 1]
```diff
@@ -473,34 +473,56 @@ impl Rav1dFrameType {
...
+    pub const FILM: Self = Self(8);
+    pub const BT2020: Self = Self(9);
+    pub const XYZ: Self = Self(10);
+    pub const SMPTE431: Self = Self(11);
+    pub const SMPTE432: Self = Self(12);
+    pub const EBU3213: Self = Self(22);
+
+    const fn to_dav1d(self) -> Dav1dTransferCharacteristics {
+        self.0 as Dav1dTransferCharacteristics
+    }
```
Comment (randomPoison) [inline include/dav1d/headers.rs:510]: `Dav1dTransferCharacteristics`? Shouldn't this be converting the value to `Dav1dColorPrimaries`?

**Review by randomPoison — APPROVED** [review]
Looks good, but I think your last rebase pulled in a bunch of extra commits. Once you fix that I can merge this :)

**Comment by folkertdev** [comment]
this merges into https://github.com/memorysafety/rav1d/pull/892 (so, not `main`) so it's probably easier to merge that one first

**Comment by randomPoison** [comment]
Okay, that PR is merged.

**Pull request merged**